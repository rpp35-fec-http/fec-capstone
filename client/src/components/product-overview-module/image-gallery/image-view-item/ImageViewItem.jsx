import React from 'react';
import Box from '@mui/material/Box';

const ImageViewItem = ({ image, mainImage, handleChildZoom }) => {
  //handles conditionally changing pointer to zoom if main image is hovered over and clicked
  const handleClick = () => {
    handleChildZoom();
  };
  return (
    <>
      {mainImage ? (
        <Box
          sx={{ cursor: 'zoom-in' }}
          component='img'
          src={image}
          onClick={handleClick}
        ></Box>
      ) : (
        <Box component='img' src={image}></Box>
      )}
    </>
  );
  // return <img src={img}></img>;
};

export default ImageViewItem;
