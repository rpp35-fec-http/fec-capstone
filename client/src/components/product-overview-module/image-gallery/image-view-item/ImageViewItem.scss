@use '../../../App.scss' as *;

//checkbox is used to cover the elements under it and act as as way to mark
//whether view should be scaled or enlarged
input[type='checkbox'] {
  position: absolute;
  cursor: nesw-resize;
  @include set-size(100%);
  z-index: 3;
  opacity: 0;
}

input[type='checkbox']:checked {
  cursor: sw-resize;
}

input[type='checkbox']:checked ~ div.enlarged {
  transform: scale(2.5);
}

.enlarged {
  @include set-size(80%, 100vw);
  position: absolute;
  z-index: 1;
}

.enlarged-view-btn {
  position: absolute;
  z-index: 10;
  left: 90%;
  margin: 0;
  padding: 0;
}

.default {
  cursor: zoom-in;
  @include set-size(70%, 50vw);
  max-height: 75vh;
}

img {
  object-fit: cover;
  width: 100%;
}
