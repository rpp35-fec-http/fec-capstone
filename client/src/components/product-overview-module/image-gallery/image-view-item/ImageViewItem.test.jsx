import renderer from 'react-test-renderer';
import ImageViewItem from './ImageViewItem.jsx';

it('renders correctly', () => {
  const tree = renderer.create(<ImageViewItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
