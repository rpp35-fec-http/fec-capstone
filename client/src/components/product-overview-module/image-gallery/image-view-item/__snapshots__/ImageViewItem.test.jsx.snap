// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="default"
  onClick={[Function]}
>
  <img
    alt="main product image "
    className="default-img"
    loading="lazy"
    src="test-file-stub"
  />
</div>
`;
